### ISSUE TITLE
Configuring recursive serialization in one-to-many relationships

### ISSUE THREAD

**Issue by mdchoate** [jfinkels/flask-restless#403]
I'm using flask-restless with Ember.js, and it seems that the way that flask-restless handles 'deep' relations doesn't play well with how Ember does it. I'm new to both frameworks, so I apologize if this is: obvious / somewhere in the docs / not a problem for anyone else. 

If an object has a one-to-many relationship, Ember expects the returned JSON to look like this (made-up example):

```
{
  'classes': [
    {
      'id': 1,
      'title': 'Algebra',
      'students': [
        4,
        11
      ]
    },
    {
      'id': 2, 
      'title': 'History',
      'students': [
        9,
        15
      ]
    } 
  ]
}
```

Where 'students' is a list of primary keys for the objects that each 'class' is related to. Ember would then use those ids to fetch the relevant objects from the API. *But,* flask-restless recursively expands the ids into the full objects before sending the response, so that 'students' is a list like `[ {'id':4, 'name':'Alice'}, {'id':11, 'name':'Bruce'} ]`. Ideally, there would be an option to simply return the un-expanded list of ids as JSON. 

As far as I can tell, this behavior isn't configurable? I found the relevant code in the to_json function in helpers.py. I was able to "fix" it for my own use, breaking the recursive behavior by inserting these two lines at helpers.py:363 - 

```python
result[relation] = [val.id for val in relatedvalue]
continue
```

...but this is obviously a bad idea. Is there any other way to modify this behavior that I missed? Would it actually be useful to anyone else to make it configurable? Again, I'm relatively new to these frameworks (and OSS in general), so I'm probably missing something - but I've found flask-restless and Ember.js to be a good match for each other in every case except this.

**Comment by jfinkels** [comment]
This is a problem we've had for a long time because I was never really sure of the best way to solve it. I do think now that I'm a little older and wiser that just providing the IDs (or URLs) of instances of the relation is the correct solution.

I'm closing this because it is essentially the same as #168. For more comments, please continue the discussion there.

**Comment by lustlost** [comment]
I also encountered this problem, this problem in the new version solution?

**Comment by jfinkels** [comment]
I have rewritten Flask-Restless to implement the JSON API protocol, so related resources are by default serialized as resource identifier objects, `{type: ..., id: ...}` instead of full resource objects; for more information, see the "[Fetching resources and relationships][1]" section of the documentation for the master branch. Also, we now correctly support model-specific serialization, reported as issue #475 and fixed in pull request #493.

[1]: http://flask-restless.readthedocs.org/en/latest/fetching.html
